File: Modul 6 Sem 1/C/PRAK604-2410817110005-MuhammadRizqanKarim.c
#include <stdio.h> 
#include <string.h> 

void decipher(char kata[], char kode[]){
    if (strlen(kata) != strlen(kode)) {
        printf("Panjang kalimat berbeda, pesan palsu");
        return;
    }
    int word = strlen(kata);
    int bintang = 0;
    int pagar = 0;
    for (int i = 0; i < word; i++) {
        if (kata[i] == kode[i]  && kata[i] != ' ') {
            printf("*");
            bintang++;
        }
        else if (kata[i] != kode[i] && kata[i] != ' ' || kode[i] != ' '){
            printf("#");
            pagar++;
        }
        else {
            printf(" ");
        }
    }
    printf("\n* = %d", bintang);
    printf("\n# = %d", pagar);
    if (bintang >= pagar) {
        printf("\nPesan Asli");
    }
    else {
        printf("\nPesan Palsu");
    }
}
int main() { 
    char kata[100], kode[100];
    
    scanf("%[^\n]", kata);
    getchar();
    scanf("%[^\n]", kode);
    getchar();

    decipher(kata, kode);
    return 0;
}

File: Modul 6 Sem 1/Phython/PRAK604-2410817110005-MuhammadRizqanKarim.py
def decipher(kata, kode):
    if len(kata) != len(kode):
        print("Panjang kalimat berbeda, pesan palsu")
        return

    bintang = 0
    pagar = 0

    for i in range(len(kata)):
        if kata[i] == kode[i] and kata[i] != ' ':
            print("*", end="")
            bintang += 1
        elif kata[i] != kode[i] and (kata[i] != ' ' or kode[i] != ' '):
            print("#", end="")
            pagar += 1
        else:
            print(" ", end="")

    print(f"\n* = {bintang}")
    print(f"# = {pagar}")

    if bintang >= pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

if __name__ == "__main__":
    kata = input("Masukkan kata: ")
    kode = input("Masukkan kode: ")

    decipher(kata, kode)